crates/interledger-stream/src/error.rs

```diff
@@ -1,12 +1,35 @@
+use interledger_packet::{AddressError, ErrorCode, PacketTypeError as IlpPacketTypeError};
 /// Stream Errors
+use std::str::Utf8Error;
+
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error("Error connecting: {0}")]
-    ConnectionError(String),
-    #[error("Error polling: {0}")]
-    PollError(String),
-    #[error("Error polling: {0}")]
-    SendMoneyError(String),
-    #[error("Error maximum time exceeded: {0}")]
-    TimeoutError(String),
+    #[error("Terminating payment since too many packets are rejected ({0} packets fulfilled, {1} packets rejected)")]
+    PaymentFailFast(u64, u64),
+    #[error("Packet was rejected with ErrorCode: {0}")]
+    SendMoneyPacket(ErrorCode),
+    #[error("Error maximum time exceeded since last fulfill")]
```

Comment:
```suggestion
    #[error("Maximum time exceeded since last fulfill")]
```
Or are these variants used in non-error contexts? Note the changed prefixes altogether, in case this was used in the stream api/rejects/at some level that someone might get access to comparing the server sent messages and hardcode these.

Reply:
This are used in error context. 
Have kept the message the same as this is a refactor